File: activity/src/main/java/com/review/activity/TwoActivity.java
package com.review.activity;

import android.app.Activity;
import android.os.Bundle;

/**
 * @author 张全
 */

public class TwoActivity extends Activity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        System.out.println("Two Activity");
    }
}

File: apk_shirking/src/main/java/com/example/apk_reduce/TwoActivity.kt
package com.example.apk_reduce

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

/**
 * 只要在manifest.xml中定义了
 * 不管是否使用 代码都会保存
 * @author zhangquan
 */
class TwoActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val layout = resources.getIdentifier("reflect_layout", "layout", packageName)
        setContentView(layout)
        method1()
    }

    fun method1() {

    }

    fun unusedMethod() {

    }
}